src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -175,25 +175,32 @@ CheckForDistributedDeadlocks(void)
 			{
 				TransactionNode *currentNode =
 					(TransactionNode *) lfirst(participantTransactionCell);
+				bool transactionAssociatedWithProc = false;
 
 				TimestampTz youngestTimestamp =
 					youngestTransaction->transactionId.timestamp;
 				TimestampTz currentTimestamp = currentNode->transactionId.timestamp;
 
-				AssociateDistributedTransactionWithBackendProc(currentNode);
+				transactionAssociatedWithProc =
+					AssociateDistributedTransactionWithBackendProc(currentNode);
 
 				LogTransactionNode(currentNode);
 
-				if (currentNode->transactionId.initiatorNodeIdentifier ==
-					GetLocalGroupId() &&
+				if (transactionAssociatedWithProc &&
 					timestamptz_cmp_internal(currentTimestamp, youngestTimestamp) == 1)
 				{
 					youngestTransaction = currentNode;
 				}
 			}
 
-			/* we should find the backend */
-			Assert(youngestTransaction->initiatorProc != NULL);
+			/*
+			 * There was a deadlock but the process already gone,
+			 * continue searching for the other distributed deadlocks.
+			 */
+			if (youngestTransaction->initiatorProc == NULL)
```

Comment:
`initiatorProc` may or may not get set on the original `transactionNode` in `AssociateDistributedTransactionWithBackendProc`, which makes this code a bit hard to follow.

What if we initialise `youngestTransaction` as `NULL` at the start and only set it if  `AssociateDistributedTransactionWithBackendProc` returned true? Maybe also rename it to `youngestRunningTransaction`.